pull_seqr_pedigrees.py

```diff
@@ -0,0 +1,92 @@
+import argparse
+import getpass
+import logging
+
+import requests
+
+logging.basicConfig(
+    format="%(asctime)s): %(message)s", datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def pull_project_peds(email: str, projects: set):
+    """
+    Uses the seqr API to retrieve a list of projects' pedigrees. Writes a single pedigree,
+    'seqr_pedigrees.txt', using the retrieved pedigrees. Failing projects are written
+    to 'projects_not_pulled.txt'
+    :param email: email used for seqr login
+    : param projects:  set of seqr project GUIDs
+    """
+    with requests.Session() as s:
+        p = s.post(
+            "https://seqr.broadinstitute.org/api/login",
+            json={"email": email, "password": getpass.getpass(),},
+        )
+        with open("seqr_pedigrees.txt", "w") as final_ped, open("projects_not_pulled.txt", "w") as errors:
+            final_ped.write(
+                "Project_GUID\tFamily_ID\tIndividual_ID\tPaternal_ID\tMaternal_ID\tSex\n"
+            )
+            for project_guid in projects:
+                project_guid = project_guid.rstrip()
+                r = s.get(
+                    f"https://seqr.broadinstitute.org/api/project/{project_guid}/details"
+                )
+
+                if r.status_code != requests.codes["ok"]:
+                    logger.info(f"Could not find {project_guid} in seqr")
+                    errors.write(f"{project_guid}\n")
+                else:
+                    fams = r.json()["familiesByGuid"]
+                    fams_dict = {}
+                    for fam in fams.values():
+                        fams_dict[fam["familyGuid"]] = fam["familyId"]
+                    inds = r.json()["individualsByGuid"]
+                    for ind in inds.values():
+                        family_id = fams_dict[ind["familyGuid"]]
+                        project_guid = ind["projectGuid"]
+                        individual_id = ind["individualId"]
+                        paternal_id = ind["paternalId"]
+                        maternal_id = ind["maternalId"]
+                        sex = ind["sex"]
```

Comment:
since these variables are only used once and their names aren't adding to readability, I'd just do
"\t".join([
                                    ind["projectGuid"],
                                    family_id,
                                    ind["individualId"],
                                    ind["paternalId"] or "",
                                    ind["maternalId"] or "",
                                    ind["sex"],
                                ])
Another shortcut if you prefer None to "" is 
```
"\t".join(map(str, [  ind["projectGuid"],  ... ] ))
```

Reply:
I like that second approach for keeping `None`. @klaricch , do you prefer "None" or "" when a paternal or maternal ID is missing?

Reply:
i think None is preferable